src/sdk/namespace_fs.js

```diff
@@ -513,7 +549,37 @@ class NamespaceFS {
                 if (r.common_prefix) {
                     res.common_prefixes.push(r.key);
                 } else {
-                    res.objects.push(this._get_object_info(bucket, r.key, r.stat, 'null'));
+                    const obj_info_list = [];
+                    const obj_info = this._get_object_info(bucket, r.key, r.stat, 'null', true);
+                    if (!list_versions && obj_info.delete_marker) {
```

Comment:
can this situation happen? since we are not listing objects under .versions/ and delete_marker must be under .versions/

Reply:
This is incase of `list_objects`. When the latest object has `delete_marker`, then `list_objects` wont expect it to list the object which has `delete_marker`

Reply:
but on list_objects we won't read .versions folder and delete_marker even if it's the latest will stay under .versions

Reply:
So the latest object will be moved under .versions once deleted ?

I am bit confused here

Reply:
on deletion of latest version (version id not specified) -
1. latest version moves to .versions/
2. new delete marker is created under .versions/ and will be the latest version

On deletion of latets version (version id specified) - 
1. the version itself will be deleted
2. If second latest version exists and not a delete marker it will move outside from .versions/ to be the latest version

Reply:
Makes sense. Removed explicit assignment `latest = false`